test/src/main/java/ai/timefold/solver/test/impl/score/stream/DefaultSingleConstraintAssertion.java

```diff
@@ -176,28 +189,28 @@ private void assertJustification(String message, ConstraintJustification... just
             throw new AssertionError(assertionMessage);
         }
 
-        List<Object> noneMatchedList = new LinkedList<>();
-        List<Object> invalidMatchList = new LinkedList<>();
+        List<Object> expectedNotFound = new LinkedList<>();
```

Comment:
Let's use `ArrayList` exclusively. `LinkedList` theoretically has some advantages, but practically, it has none.

This doesn't exactly match our use case, but it does a good job of explaining why the reasons people like LinkedList are actually kinda false:

https://kjellkod.wordpress.com/2012/02/25/why-you-should-never-ever-ever-use-linked-list-in-your-code-again/